bottlecap/src/proc/mod.rs

```diff
@@ -0,0 +1,84 @@
+pub mod constants;
+
+use std::{
+    fs::File,
+    io::{self, BufRead},
+};
+
+use constants::{LAMDBA_NETWORK_INTERFACE, PROC_NET_DEV_PATH};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct NetworkData {
+    pub rx_bytes: f64,
+    pub tx_bytes: f64,
+}
+
+pub fn get_network_data() -> Result<NetworkData, io::Error> {
+    get_network_data_from_path(PROC_NET_DEV_PATH)
+}
+
+fn get_network_data_from_path(path: &str) -> Result<NetworkData, io::Error> {
+    let file = File::open(path)?;
+    let reader = io::BufReader::new(file);
+
+    for line in reader.lines() {
+        let line = line?;
+        let mut values = line.split_whitespace();
+
+        // Check for the line containing lambda network data by interface name
+        if let Some(interface_name) = values.next() {
+            if interface_name.starts_with(LAMDBA_NETWORK_INTERFACE) {
```

Comment:
I think you can simplify it more by chaining this to something like
```suggestion
       if values.next().map_or(false, |interface_name| 
            interface_name.starts_with(LAMDBA_NETWORK_INTERFACE)) {
}
```